recipes-connectivity/networkmanager/networkmanager-1.4.4/xc-nutty-network.patch

```diff
@@ -0,0 +1,366 @@
++		if (oxt_nlmsg)
++			do_delete_object (platform, &oxt_obj_id, oxt_nlmsg);
++	}
++
+ 	if (metric == 0) {
+ 		/* Deleting an IPv4 route with metric 0 does not only delete an exectly matching route.
+ 		 * If no route with metric 0 exists, it might delete another route to the same destination.
+@@ -5841,6 +5902,7 @@ ip4_route_delete (NMPlatform *platform,
+ 	                           0,
+ 	                           AF_INET,
+ 	                           ifindex,
++				   RT_TABLE_MAIN,
+ 	                           NM_IP_CONFIG_SOURCE_UNKNOWN,
+ 	                           RT_SCOPE_NOWHERE,
+ 	                           &network,
+@@ -5867,6 +5929,7 @@ ip6_route_delete (NMPlatform *platform,
+ 	                           0,
+ 	                           AF_INET6,
+ 	                           ifindex,
++				   RT_TABLE_MAIN,
+ 	                           NM_IP_CONFIG_SOURCE_UNKNOWN,
+ 	                           RT_SCOPE_NOWHERE,
+ 	                           &network,
+--- a/src/devices/nm-device.c
++++ b/src/devices/nm-device.c
+@@ -1632,7 +1632,7 @@ carrier_changed (NMDevice *self, gboolea
+ 			nm_device_activate_stage3_ip6_start (self);
+ 
+ 		return;
+-	} else if (nm_device_get_enslaved (self) && !carrier) {
++	} else if (nm_device_get_enslaved (self) && !carrier && strncmp(priv->iface, "eth", 3)) {
+ 		/* Slaves don't deactivate when they lose carrier; for
+ 		 * bonds/teams in particular that would be actively
+ 		 * counterproductive.
+@@ -1659,7 +1659,11 @@ carrier_changed (NMDevice *self, gboolea
+ 			nm_device_update_dynamic_ip_setup (self);
+ 		}
+ 	} else {
+-		if (priv->state == NM_DEVICE_STATE_UNAVAILABLE) {
++		if (strncmp (priv->iface, "eth", 3) == 0) {
++			nm_device_queue_state (self, NM_DEVICE_STATE_DISCONNECTED,
++		        	               NM_DEVICE_STATE_REASON_CARRIER);
++		}
++		else if (priv->state == NM_DEVICE_STATE_UNAVAILABLE) {
+ 			if (nm_device_queued_state_peek (self) >= NM_DEVICE_STATE_DISCONNECTED)
+ 				nm_device_queued_state_clear (self);
+ 		} else {
+@@ -2622,6 +2626,7 @@ slave_state_changed (NMDevice *slave,
+ 	NMDevicePrivate *priv = NM_DEVICE_GET_PRIVATE (self);
+ 	gboolean release = FALSE;
+ 	gboolean configure = TRUE;
++	const char *iface = nm_device_get_iface (slave);
+ 
+ 	_LOGD (LOGD_DEVICE, "slave %s state change %d (%s) -> %d (%s)",
+ 	       nm_device_get_iface (slave),
+@@ -2648,7 +2653,7 @@ slave_state_changed (NMDevice *slave,
+ 	if (reason == NM_DEVICE_STATE_REASON_CONNECTION_ASSUMED)
+ 		configure = FALSE;
+ 
+-	if (release) {
++	if (release && strncmp (iface, "eth", 3)) {
+ 		nm_device_master_release_one_slave (self, slave, configure, reason);
+ 		/* Bridge/bond/team interfaces are left up until manually deactivated */
+ 		if (priv->slaves == NULL && priv->state == NM_DEVICE_STATE_ACTIVATED)
+--- a/src/devices/nm-device-ethernet-utils.c
++++ b/src/devices/nm-device-ethernet-utils.c
+@@ -31,6 +31,7 @@ nm_device_ethernet_utils_get_default_wir
+ 	guint j;
+ 	int i;
+ 
++	return NULL;
```

Comment:
That shortcuts a few things, so no ethernet is returned to be the "default" connection?

Reply:
Allowing the creation of 'Wired connection %d' is of no use, eth0 on discovering it as a slave for brbridged assumes a slave connection and this connection is left as is instead of being cleaned up. In fact this creates a problem when the ethernet cable is unplugged and plugged back.
On unplug/plug of ethernet cable eth0 simply adopts this connection instead of being a slave connection for brbridged.